dev/integration_tests/android_engine_test/tool/deflake.dart

```diff
@@ -0,0 +1,163 @@
+    _printUsage();
+    io.exitCode = 1;
+    return;
+  }
+
+  final bool generateInitialGolden = argResults.flag('generate-initial-golden');
+  final bool buildAppOnce = argResults.flag('build-app-once');
+  final bool verbose = argResults.flag('verbose');
+  final int runs;
+  {
+    final String rawRuns = argResults.option('runs')!;
+    final int? parsedRuns = int.tryParse(rawRuns);
+    if (parsedRuns == null || parsedRuns < 1) {
+      io.stderr.writeln('--runs must be a positive number: "$rawRuns".');
+      io.exitCode = 1;
+      return;
+    }
+    runs = parsedRuns;
+  }
+
+  final List<String> driverArgs;
+  if (buildAppOnce) {
+    io.stderr.writeln('Building initial app with "flutter build apk --debug...');
+    final io.Process proccess = await io.Process.start('flutter', <String>[
+      'build',
+      'apk',
+      '--debug',
+      testFile.path,
+    ], mode: verbose ? io.ProcessStartMode.inheritStdio : io.ProcessStartMode.normal);
+    if (await proccess.exitCode case final int exitCode when exitCode != 0) {
+      io.stderr.writeln('Failed to build (exit code = $exitCode).');
+      io.stderr.writeln(_collectStdOut(proccess));
+      io.exitCode = 1;
+      return;
+    }
+
+    // Strictly speaking, it would be better to parse stdout for:
+    // "✓ Built build/app/outputs/flutter-apk/app-debug.apk"
+    //
+    // ... _or_ specify the expected out ourselves and rely on that.
+    driverArgs = <String>[
+      'drive',
+      '--use-application-binary',
+      p.join('build', 'app', 'outputs', 'flutter-apk', 'app-debug.apk'),
+      testFile.path,
+    ];
+  } else {
+    // I can't imagine wanting to do this, but here is the option anyway!
+    driverArgs = <String>['drive', testFile.path];
+  }
+
+  Future<bool> runDriverTest({Map<String, String>? environment}) async {
+    final io.Process proccess = await io.Process.start(
+      'flutter',
+      driverArgs,
+      mode: verbose ? io.ProcessStartMode.inheritStdio : io.ProcessStartMode.normal,
+      environment: environment,
+    );
+    if (await proccess.exitCode case final int exitCode when exitCode != 0) {
+      io.stderr.writeln('Failed to build (exit code = $exitCode).');
+      io.stderr.writeln(_collectStdOut(proccess));
+      return false;
+    }
+    return true;
+  }
+
+  // Do an initial baseline run.
+  if (generateInitialGolden) {
+    io.stderr.writeln('Generating a baseline set of golden-files...');
+    await runDriverTest(environment: <String, String>{'UPDATE_GOLDENS': '1'});
+  }
+
+  // Now run.
+  int totalFailed = 0;
+  for (int i = 0; i < runs; i++) {
+    io.stderr.writeln('RUN ${i + 1} of $runs');
+    final bool result = await runDriverTest();
+    if (!result) {
+      totalFailed++;
+      io.stderr.writeln('FAIL');
+    } else {
+      io.stderr.writeln('PASS');
+    }
+  }
+
+  io.stderr.writeln('PASSED: ${runs - totalFailed} / $runs');
+  if (totalFailed != 0) {
+    io.exitCode = 1;
+  }
+}
+
+final ArgParser _argParser =
```

Comment:
nonblocking nit: I would prefer this to be at the top of main. My inital read I didn't realize there were flags that could be passed and thought your documentation was future looking.